Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/DriverFeeUpdates/DriverFee.hs

```diff
@@ -158,15 +158,13 @@ calculateDriverFeeForDrivers Job {id, jobInfo} = withLogTag ("JobId-" <> id.getI
               coinCashLeft = maybe 0 (.coinCovertedToCashLeft) mbDriverPlan
 
           driver <- QP.findById (cast driverFee.driverId) >>= fromMaybeM (PersonDoesNotExist driverFee.driverId.getId)
-
+          let numRides = driverFee.numRides - plan.freeRideCount - if transporterConfig.considerSpecialZoneRidesForPlanCharges then 0 else driverFee.specialZoneRideCount
```

Comment:
I think it is correct because `driverFee.numRides` contains both normal + special zone rides and if the flag `considerSpecialZoneRidesForPlanCharges` is false then we need to deduct the special zone rides count.